teuthology/orchestra/remote.py

```diff
@@ -61,25 +61,24 @@ def connect(self, timeout=None):
         self.ssh = connection.connect(**args)
         return self.ssh
 
-    def reconnect(self, timeout=None):
+    def reconnect(self, timeout=None, socket_timeout=None, sleep_time=30):
         """
         Attempts to re-establish connection. Returns True for success; False
         for failure.
         """
         if self.ssh is not None:
             self.ssh.close()
         if not timeout:
-            return self._reconnect(timeout=timeout)
+            return self._reconnect(timeout=socket_timeout)
         start_time = time.time()
         elapsed_time = lambda: time.time() - start_time
         while elapsed_time() < timeout:
-            success = self._reconnect()
+            success = self._reconnect(timeout=socket_timeout)
             if success:
                 break
-            default_sleep_val = 30
```

Comment:
Can you paste the teuthology test logs here for krbd or kcephfs tests?, Also its a standard to run tests that got affected. I am still skeptical about this sleep being deleted,  the rest of the *new* kw args are not going to change existing logic.

Reply:
@vasukulkarni It is not deleted, it is moved to the signature of the method, look carefully.

Reply:
@kshtsk thanks I see that now, I am dismissing my review but please post the test results as well.

Reply:
@vasukulkarni Do you have an idea how to run krbd with kcephfs test with this patch? Do you have a live system to test it?

Reply:
@kshtsk As I read from your other comment, it looks like you dont have access to sepia, If you request the account, I can help you on running the test, its a simple cli.

Reply:
> @kshtsk As I read from your other comment, it looks like you dont have access to sepia, If you request the account, I can help you on running the test, its a simple cli.

is it so simple that it will take this change to teuthology server/worker code?

Reply:
?

Reply:
> ?

how one can be sure the patch in PR for teuthology is used for running a test? is there a testing teuthology instance installed, or we need to run a special teuthology-worker  on this code?

Reply:
> > ?
> 
> how one can be sure the patch in PR for teuthology is used for running a test? is there a testing teuthology instance installed, or we need to run a special teuthology-worker on this code?

I did a git fetch since it was in your repo and pushed to 'wip-reconnect-timeout', then I used the following cli to schedule in sepia lab using your branch

./teuthology-suite -n 10 -s kcephfs -k distro -c luminous --ceph-repo http://github.com/ceph/ceph.git   --suite-repo http://github.com/ceph/ceph.git -m mira -t wip-reconnect-timeout -p 90  -l 4

Results using your branch should show here: http://pulpito.ceph.com/vasu-2019-02-13_22:38:31-kcephfs-luminous-distro-basic-mira/

Reply:
> > > ?
> > 
> > 
> > how one can be sure the patch in PR for teuthology is used for running a test? is there a testing teuthology instance installed, or we need to run a special teuthology-worker on this code?
> 
> I did a git fetch since it was in your repo and pushed to 'wip-reconnect-timeout', then I used the following cli to schedule in sepia lab using your branch
> 
> ./teuthology-suite -n 10 -s kcephfs -k distro -c luminous --ceph-repo [http://github.com/ceph/ceph.git](https://github.com/ceph/ceph.git) --suite-repo [http://github.com/ceph/ceph.git](https://github.com/ceph/ceph.git) -m mira -t wip-reconnect-timeout -p 90 -l 4
> 
> Results using your branch should show here: http://pulpito.ceph.com/vasu-2019-02-13_22:38:31-kcephfs-luminous-distro-basic-mira/

Thanks for scheduling the tests for me, I can see the correct sha1 in the beginning of the logs. There are failing jobs, could you please confirm those failures not related to this change.

```
2019-02-13T22:58:54.731 INFO:teuthology.orchestra.run.mira083:> sudo hostname mira083
2019-02-13T22:58:54.805 INFO:teuthology.provision.fog.mira083:Deploy complete!
2019-02-13T22:58:54.806 DEBUG:teuthology.parallel:result is None
2019-02-13T22:58:56.297 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira056.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:58:56.516 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2019-02-13T22:59:02.518 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira056.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:59:02.656 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira041.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:59:02.922 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:02.922 INFO:teuthology.orchestra.run.mira041:> hostname
2019-02-13T22:59:04.178 INFO:teuthology.orchestra.run.mira041.stdout:mira041.front.sepia.ceph.com
2019-02-13T22:59:04.178 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.178 INFO:teuthology.orchestra.run.mira041:> grep mira041.front.sepia.ceph.com /etc/hosts
2019-02-13T22:59:04.246 INFO:teuthology.orchestra.run.mira041.stdout:172.21.5.132 mira041.front.sepia.ceph.com mira041
2019-02-13T22:59:04.247 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.247 INFO:teuthology.orchestra.run.mira041:> sudo hostname mira041
2019-02-13T22:59:04.401 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.401 INFO:teuthology.orchestra.run.mira041:> sudo sed -i -e 's/mira041.front.sepia.ceph.com/mira041/g' /etc/hosts
2019-02-13T22:59:04.411 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.412 INFO:teuthology.orchestra.run.mira041:> sudo sed -i -e 's/172.21.5.132/172.21.5.132/g' /etc/hosts
2019-02-13T22:59:04.486 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.486 INFO:teuthology.orchestra.run.mira041:> sudo sed -i -e 's/mira041.front.sepia.ceph.com/mira041/g' /etc/hostname
2019-02-13T22:59:04.560 INFO:teuthology.orchestra.run.mira041:Running:
2019-02-13T22:59:04.560 INFO:teuthology.orchestra.run.mira041:> sudo hostname mira041
2019-02-13T22:59:04.632 INFO:teuthology.provision.fog.mira041:Deploy complete!
2019-02-13T22:59:04.634 DEBUG:teuthology.parallel:result is None
2019-02-13T22:59:05.570 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2019-02-13T22:59:11.575 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira056.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:59:12.229 INFO:teuthology.orchestra.run.mira056:Running:
2019-02-13T22:59:12.229 INFO:teuthology.orchestra.run.mira056:> hostname
2019-02-13T22:59:14.043 INFO:teuthology.orchestra.run.mira056.stdout:mira056.front.sepia.ceph.com
2019-02-13T22:59:14.043 INFO:teuthology.orchestra.run.mira056:Running:
2019-02-13T22:59:14.043 INFO:teuthology.orchestra.run.mira056:> grep mira056.front.sepia.ceph.com /etc/hosts
2019-02-13T22:59:14.114 INFO:teuthology.orchestra.run.mira056.stdout:172.21.6.138 mira056.front.sepia.ceph.com mira056
2019-02-13T22:59:14.115 INFO:teuthology.orchestra.run.mira056:Running:
```
From this looks it reconnect works as before, doesn't it?

Reply:
The failures are unrelated, this part does reconnection for sure

```
2019-02-13T22:56:11.948 INFO:teuthology.misc:Re-opening connections...
2019-02-13T22:56:11.948 INFO:teuthology.misc:trying to connect to ubuntu@mira111.front.sepia.ceph.com
2019-02-13T22:56:11.949 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira111.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:56:12.136 INFO:teuthology.orchestra.run.mira111:Running:
2019-02-13T22:56:12.136 INFO:teuthology.orchestra.run.mira111:> true
2019-02-13T22:56:12.708 INFO:teuthology.misc:trying to connect to ubuntu@mira052.front.sepia.ceph.com
2019-02-13T22:56:12.710 DEBUG:teuthology.orchestra.connection:{'username': 'ubuntu', 'hostname': 'mira052.front.sepia.ceph.com', 'timeout': 60}
2019-02-13T22:56:12.828 INFO:teuthology.orchestra.run.mira052:Running:
2019-02-13T22:56:12.828 INFO:teuthology.orchestra.run.mira052:> true
2019-
```